File: e2e/selenide-selenoid-junit5/src/test/java/page/CreateTaskForm.java
package page;

import com.codeborne.selenide.SelenideElement;
import data.TaskPriority;
import data.TaskStatus;

import static com.codeborne.selenide.Selenide.$;

public class CreateTaskForm {
    public SelenideElement createButton = $("[data-testid='create-button']");
    public SelenideElement titleField = $("[data-testid='create-task-title-input']");
    public SelenideElement descriptionField = $("#description");
    public SelenideElement statusDropdown = $("[data-testid='status-dropdown']");
    public SelenideElement statusDropdownLabel = statusDropdown.$(".p-dropdown-label");
    public SelenideElement priorityDropdown = $("[data-testid='priority-dropdown']");
    public SelenideElement priorityDropdownLabel = priorityDropdown.$(".p-dropdown-label");

    public SelenideElement statusOption(TaskStatus taskStatus) {
        return $("[data-testid='status-dropdown-option-" + taskStatus + "']");
    }

    public SelenideElement priorityOption(TaskPriority taskPriority) {
        return $("[data-testid='priority-dropdown-option-" + taskPriority + "']");
    }
}

File: demo-ios/DemoUITests/Interaction/Page/CreateTaskForm.swift
import XCTest

final class CreateTaskForm: XCUIPage {
    func createButton() -> XCUIElement {
        element("create-button")
    }

    func titleField() -> XCUIElement {
        element("create-task-title-input")
    }

    func descriptionField() -> XCUIElement {
        element("task-description-input")
    }

    func statusDropdown() -> XCUIElement {
        element("status-dropdown")
    }

    func priorityDropdown() -> XCUIElement {
        element("priority-dropdown")
    }

    func statusOption(_ status: TaskStatus) -> XCUIElement {
        element("status-dropdown-option-\(status.rawValue)")
    }

    func priorityOption(_ priority: TaskPriority) -> XCUIElement {
        element("priority-dropdown-option-\(priority.rawValue)")
    }
}
